modules/packages/LinearAlgebra.chpl

```diff
@@ -1678,6 +1678,49 @@ module Sparse {
     }
     return Y;
   }
+  
+  /* Sparse Matrix-vector multiplication with distributed support */
+  proc _sparsematvecMult(const ref A: [?Adom] ?eltType, const ref X: [?Xdom] eltType) where isSparseArr(A) {
+    var Y: [Xdom] eltType;
+    var locks$: [LocaleSpace] sync bool;
+    coforall l in Locales do on l {
+      const localDomain = A.localSubdomain();
+      var (low_i, low_j) = localDomain.low;
+      var (high_i, high_j) = localDomain.high;
+      var rowResults: [low_i..high_i] eltType = 0;
+
+      forall (i,j) in localDomain {
+          rowResults[i] += A[i,j] * X[j];
+      }
+      
+      // Writing result
+      // This part assumes 0-based indexing of Adom
+      const numRowsPerSynch = localDomain.dim(1).size / A.domain.dist.targetLocales().shape(2);
+      const targetLoc = A.domain.dist.targetLocales();
+      var numLocsInRow = targetLoc.shape(2);
+      var rowid, colid: int;
+      for (ind, Loc) in zip(targetLoc.domain, targetLoc) {
```

Comment:
Loc -> loc

Reply:
I think this loop deserves a small comment. This loop is trying to find where the locale is in the targetLocales array in a rather naive way. There must be a support for that in the distributions to get that without having to do this manually.